### PR TITLE
Ana/fix loading logic for data properties

### LINKED ISSUE #180: Lunie Light: loading logic broken for delegations and balances
Since we don't have Apollo anymore, we are displaying the Loading... div based on `balances` and `delegations` length. The problem is that this length is already 0 in `state`.

So for accounts without delegations the Loading... div appears eternally. I think the same is happening for balances, although it should return a `LunieCoin` with `amount: 0`. I will investigate that.

Simple solution: set `balances` and `delegations` and so on to `null` in `state`

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #180

Good to go! 

I will open another PR returning an `amount: 0` balance for accounts with no balance

image

### CODE REVIEW

**Comment by jbibla** [comment]
i don't think this works

Screen Shot 2020-11-11 at 10 49 44 AM

File: components/portfolio/DelegationsOverview.vue [hunk 1]
```diff
@@ -16,7 +16,15 @@
           show-on-mobile="expectedReturns"
         />
       </div>
-      <div v-if="!delegations.length && !balances.length" class="loading-row">
+      <div
+        v-if="
+          !delegations.length &&
+          !balances.length &&
```
Comment (faboweb) [inline components/portfolio/DelegationsOverview.vue:22]:
```suggestion
```

File: components/portfolio/TableBalances.vue [hunk 2]
```diff
@@ -5,7 +5,9 @@
     <div class="table-cell title available">Available</div>
     <div class="table-cell title actions"></div>
 
-    <div v-if="!balances.length" class="loading-row left">Loading...</div>
+    <div v-if="!balances.length && !balancesLoaded" class="loading-row left">
```
Comment (faboweb) [inline components/portfolio/TableBalances.vue:8]:
```suggestion
    <div v-if="!balancesLoaded" class="loading-row left">
```

File: pages/validators/_address.vue [hunk 3]
```diff
@@ -1,6 +1,8 @@
 <template>
   <div>
-    <div v-if="!validators.length" class="loading-row">Loading...</div>
+    <div v-if="!validators.length && !validatorsLoaded" class="loading-row">
```
Comment (faboweb) [inline pages/validators/_address.vue:3]:
```suggestion
    <div v-if="!validatorsLoaded" class="loading-row">
```

File: store/data.js [hunk 4]
```diff
@@ -8,9 +8,13 @@ export const state = () => ({
   balances: [],
   balancesLoaded: undefined,
   rewards: [],
+  rewardsLoaded: undefined,
   delegations: [],
+  delegationsLoaded: undefined,
   undelegations: [],
+  undelegationsLoaded: undefined,
   validators: [],
+  validatorsLoaded: undefined,
```
Comment (faboweb) [inline store/data.js:17]:
```suggestion
  balancesLoaded: false,
  rewards: [],
  rewardsLoaded: false,
  delegations: [],
  delegationsLoaded: false,
  undelegations: [],
  undelegationsLoaded: false,
  validators: [],
  validatorsLoaded: false,
```

**Pull request merged**